liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -366,10 +368,20 @@ protected void doExecute() throws Exception {
                         }
                     } else if (directory.startsWith(configDirectory.toPath())) {  // config files                                                                    
                         if (fileChanged.exists() && (event.kind() == StandardWatchEventKinds.ENTRY_MODIFY ||  event.kind() == StandardWatchEventKinds.ENTRY_CREATE)) {
-                            copyFile(fileChanged, configDirectory, serverDirectory);
+                            // config directory does not exist, refresh only config file
+                            if (noConfigDir && fileChanged.getAbsolutePath().endsWith(configFile.getName())){ 
```

Comment:
Since the contents of the if and else if statements are the same, they can be combined to the equivalent statement:
`noConfigDir && fileChanged.getAbsolutePath().endsWith(configFile.getName()) || !noConfigDir`
which can be further simplified to:
`!noConfigDir || fileChanged.getAbsolutePath().endsWith(configFile.getName())`